lib/Elastica/Request.php

```diff
@@ -29,10 +31,11 @@ class Request extends Param
      * @param array      $data       OPTIONAL Data array
      * @param array      $query      OPTIONAL Query params
      * @param Connection $connection
+     * @param string     $contentType Content-Type sent with this request
      *
      * @return \Elastica\Request OPTIONAL Connection object
      */
-    public function __construct($path, $method = self::GET, $data = [], array $query = [], Connection $connection = null)
+    public function __construct($path, $method = self::GET, $data = [], array $query = [], Connection $connection = null, $contentType = self::DEFAULT_CONTENT_TYPE )
```

Comment:
Ideally we should no longer accept a String as the $data object.
Giving an array it'll be unambiguously 'application/json' but giving a string we don't know how to decide between json and the bulk format.
Creating a new class like RequestBody and enforce clients to use it when they plan to send strings directly might be a cleaner approach imo.